frame/ethereum/src/lib.rs

```diff
@@ -296,4 +342,43 @@ impl<T: Trait> Module<T> {
 
 		Pending::append((transaction, status, receipt));
 	}
+
+	fn handle_exec<R>(res: (ExitReason, R, U256)) -> Result<(ExitReason, R, U256), Error<T>> {
+		match res.0 {
+			ExitReason::Succeed(_s) => { Ok(res) },
+			ExitReason::Error(e) => { Err(Self::parse_exit_error(e)) },
+			ExitReason::Revert(e) => {
+				match e {
+					ExitRevert::Reverted => { Err(Error::<T>::Reverted) },
+				}
+			},
+			ExitReason::Fatal(e) => {
+				match e {
+					ExitFatal::NotSupported => { Err(Error::<T>::NotSupported) },
+					ExitFatal::UnhandledInterrupt => { Err(Error::<T>::UnhandledInterrupt) },
+					ExitFatal::CallErrorAsFatal(e_error) => { Err(Self::parse_exit_error(e_error)) },
+					ExitFatal::Other(_s) => { Err(Error::<T>::ExitFatalOther) },
+				}
+			},
+		}
+	}
+
+	fn parse_exit_error(exit_error: ExitError) -> Error<T> {
+		match exit_error {
+			ExitError::StackUnderflow => { return Error::<T>::StackUnderflow },
```

Comment:
Just a styling grumble.
```suggestion
			ExitError::StackUnderflow => Error::<T>::StackUnderflow,
```

And all rest of the lines.

Reply:
will change!